python/plugins/processing/core/parameters.py

```diff
@@ -1186,15 +1190,17 @@ def getAsScriptCode(self):
         param_type = ''
         if self.optional:
             param_type += 'optional '
-        param_type += 'string'
-        return '##' + self.name + '=' + param_type + self.default
+        param_type += 'string '
+        return '##' + self.name + '=' + param_type + str(self.default)
 
     @classmethod
     def fromScriptCode(self, line):
         isOptional, name, definition = _splitParameterOptions(line)
         descName = _createDescriptiveName(name)
         if definition.lower().strip().startswith('string'):
-            default = definition.strip()[len('string') + 1:]
+            default = definition.strip()[len('string') + 1:] or None
+            if default == 'None':
+                default = None
```

Comment:
@alexbruy with this approach, we don't have a possibility to save the string literal `'None'` in here. Do you think the exported parameter should be quoted?